airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/file_transfer.py

```diff
@@ -0,0 +1,31 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+import logging
+import os
+from typing import Any, Dict, Iterable
+
+from airbyte_cdk.sources.file_based.config.file_based_stream_config import FileBasedStreamConfig
+from airbyte_cdk.sources.file_based.file_based_stream_reader import AbstractFileBasedStreamReader, FileReadMode
+from airbyte_cdk.sources.file_based.remote_file import RemoteFile
+
+AIRBYTE_STAGING_DIRECTORY = os.getenv("AIRBYTE_STAGING_DIRECTORY", "/staging/files")
+DEFAULT_LOCAL_DIRECTORY = "/tmp/airbyte-file-transfer"
+
+
+class FileTransfer:
+    def __init__(self) -> None:
+        self._local_directory = AIRBYTE_STAGING_DIRECTORY if os.path.exists(AIRBYTE_STAGING_DIRECTORY) else DEFAULT_LOCAL_DIRECTORY
+
+    def get_file(
+        self,
+        config: FileBasedStreamConfig,
+        file: RemoteFile,
+        stream_reader: AbstractFileBasedStreamReader,
+        logger: logging.Logger,
+    ) -> Iterable[Dict[str, Any]]:
+        try:
+            yield stream_reader.get_file(file=file, local_directory=self._local_directory, logger=logger)
```

Comment:
Implementation sample for source-s3 [here](https://github.com/airbytehq/airbyte/pull/48346/files#diff-b5e8d7cf7e79880dd2e77408410825335670decbc45f4960d2cf07ac323bae86).

Reply:
Manual testing in [this workspace](https://cloud.airbyte.com/workspaces/166f3e04-9021-4f63-a963-a43abb548cec/connections/d7e111cf-1ddc-41bc-8b1a-34bf435c3f2f/timeline) seems be working fine for sync > 5GB :)